### PR TITLE
[openstack_*][ubuntu] Update paths and collections

### PR DESCRIPTION
**Pull request by arif-ali**
Update the apache plugin to only collect the defaults and update the openstack bits to collect specifics for their service

Closes: #2525

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [ ] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

**Comment by arif-ali** [comment]
@TurboTurtle from the RH perspective do you know if manila has logs in `/var/log/httpd`?

**Comment by arif-ali** [comment]
I am opening this up for review, for targeting 4.5.5; as it would be good to get feedback from other canonical folks

**Review by TurboTurtle — APPROVED** [review]
LGTM

File: sos/report/plugins/openstack_nova.py [hunk 1]
```diff
@@ -92,6 +92,8 @@ def setup(self):
         if self.get_option("all_logs"):
             self.add_copy_spec([
                 "/var/log/nova/",
+                "/var/log/{}*/nova*".format(self.apachepkg),
+                "/var/log/{}*/placement*".format(self.apachepkg),
```
Comment (pmoravec) [inline sos/report/plugins/openstack_nova.py:96]: We collect the same `copy_spec` also in `openstack_placement` (same applies to `"/var/log/{}*/placement*.log".format(self.apachepkg),` few lines below). Two plugins can't collect the same `copy_spec` (I guess you know the reason).

Are we sure the two plugins wont be enabled concurrently any time? Or shouldnt we collect the `placement` logs just in `openstack_placement` plugin?

**Comment by pmoravec** [comment]
ACK up to the one concern.

File: sos/report/plugins/openstack_nova.py (hunk 1 above)
Comment (arif-ali) [reply sos/report/plugins/openstack_nova.py:96]: On the Ubuntu platforms and how it is deployed, it is typically on 2 separate hosts. But, I brought these in from the RedHat section primarily.

But, it is technically possible of that happening where you have both the nova and placement on the same host.

I will re-evaluate this later

**Review by pmoravec — APPROVED** [review]

**Comment by arif-ali** [comment]
@TurboTurtle @pmoravec one thing that really threw me he on the apache2 plugin was the fact that line 48 mentions `subdir`, this in-fact is a wrong thing to label. This will exclude files or dirs under the `/var/log/<apachepkg>/`

I think, that should be renamed or refactored, maybe in a different PR

**Comment by TurboTurtle** [comment]
> @TurboTurtle @pmoravec one thing that really threw me here on the `apache2` plugin was the fact that line 48 mentions `subdir`, this in-fact is a wrong thing to label. This will exclude files or dirs under the `/var/log/<apachepkg>/`
> 
> I think, that should be renamed or refactored, maybe in a different PR

That's the intent though...so that the `apache` plugin doesn't collect them, as these should be collected by the `openstack_*` plugins. The forbidden paths here are plugin-specific, if you're seeing something else then we have a deeper problem in `Plugin`.

**Comment by arif-ali** [comment]
> > @TurboTurtle @pmoravec one thing that really threw me here on the `apache2` plugin was the fact that line 48 mentions `subdir`, this in-fact is a wrong thing to label. This will exclude files or dirs under the `/var/log/<apachepkg>/`
> > I think, that should be renamed or refactored, maybe in a different PR
> 
> That's the intent though...so that the `apache` plugin doesn't collect them, as these should be collected by the `openstack_*` plugins. The forbidden paths here are plugin-specific, if you're seeing something else then we have a deeper problem in `Plugin`.

I agree totally

I think it was the naming of `subdir` that threw me off, because it may not be a sub directory, it could be a file with that name too

**Comment by arif-ali** [comment]
yeah, I've had a +1 from a colleague already, and this will already provide something that we are not getting. Would be awesome to get through, which will give enough testing before 4.5.6 when we will potentially cutting our next `deb` release

**Pull request merged**